api/lib/opentelemetry.rb

```diff
@@ -41,10 +42,21 @@ def handle_error(exception: nil, message: nil)
     error_handler.call(exception: exception, message: message)
   end
 
+  # Register the global tracer provider.
+  #
+  # @param [TracerProvider] provider A tracer provider to register as the
+  #   global instance.
+  def tracer_provider=(provider)
+    # TODO: should this use a Mutex?
```

Comment:
What would you be concerned with here that would require the mutex, a race condition with setting the tracer_provider?

Reply:
yep. mutex added